bento/src/main/java/com/yelp/android/bento/core/ComponentGroup.java

```diff
@@ -412,6 +407,41 @@ public int findComponentOffset(@NonNull Component component) {
         return -1;
     }
 
+    /**
+     * Finds and returns the component at lowest level (leaf) that encompasses the index.
+     *
+     * @param index The index to search for.
+     * @return The lowest component in the tree.
+     */
+    @Override
+    public Component getLowestComponentAtIndex(int index) {
+        Component componentAtPos = componentAt(index);
+        Range range = rangeOf(componentAtPos);
+        if (range == null) {
+            throw new IllegalStateException("Found component, but no range associated.");
+        }
+
+        return componentAtPos.getLowestComponentAtIndex(index - range.mLower);
+    }
+
+    /**
+     * Similar to {@link #getLowestComponentAtIndex(int)}, but returns both the component and the
+     * absolute range within the controller.
+     *
+     * @param index The index to search for.
+     * @return Both a component and an absolute range over the entire controller.
+     */
+    @Override
+    public RangedValue<Component> getLowestRangeValue(int index) {
```

Comment:
This method seems very similar to `findComponentOffset` except it includes the upper bound as well.

Reply:
You aren't wrong. Are you suggesting replacing `findComponentOffset` with `getLowestRangeValue`?

Reply:
Could we combine the two methods somehow? (e.g. make a helper method that both methods call into in order to reduce some of the very similar logic)

Reply:
Or maybe just give them complimentary names (like `findComponentWithIndex` and `findIndexOfComponent` or something)